public/js/navigator-js/integration/StateViewMap.js

```diff
@@ -43,33 +45,90 @@ this.navigatorjs.integration = this.navigatorjs.integration || {};
 				state = recipeStates[j];
 
 				if (requestedState.contains(state)) {
-					viewInstance = recipe.getViewInstance();
 
-					if (viewInstance.navigatorBehaviors instanceof Array) {
-						_addViewElementToDOM(recipe);
+					_addViewElementToDOM(recipe);
+					var viewInstance = recipe.getViewInstance();
+					if (Array.isArray(viewInstance.navigatorBehaviors)) {
 						_navigator.add(viewInstance, state);
 					}
+
+				} else {
+					if (recipe.isMounted()) {
+						_removeViewElementFromDOM(recipe);
+					}
 				}
 			}
 		}
 	}
 
+	function _removeViewElementFromDOM(recipe) {
```

Comment:
Correct me if I am wrong - removing the view here only affect the DOM and does not destroy the js class. Which should be less expensive that disposing the full view. Right? 
Anyways, could we benchmark this in your demo? I would like to see the difference in the rendering time when removing / adding html node vs. display:none / ''